scripts/scraper/scrape-mdn.js

```diff
@@ -0,0 +1,133 @@
+/*
+A scraper to help migrate MDN docs to stumptown.
+
+Usage:
+
+npm run scrape-mdn <mdn-url> <destination> scrape-children
+
+<mdn-url>: A full URL to an MDN page to scrape.
+
+<destination>: Where to put the processed document. This is relative to ./content
+
+scrape-children: If 'scrape-children' is given, fetch the immediate children
+of this page and process them as well. Store the child pages under a
+subdirectory of <destination>, where the subdirectory name is the parent's name.
+
+The pages are 'processed' on the way. This means that they are:
+* converted to JSDOM where various things can be done to them, before going back to HTML
+* converted to Markdown
+* given some front matter (title and mdn_url)
+*/
+
+const unified = require('unified');
+const parse = require('rehype-parse');
+const stringify = require('remark-stringify');
+const rehype2remark = require('rehype-remark');
+
+const fetch = require("node-fetch");
+const fs = require('fs');
+const path = require('path');
+const jsdom = require('jsdom');
+const { JSDOM } = jsdom;
+
+const baseURL = 'https://developer.mozilla.org';
+
+/**
+ * Converts the HTML -> Markdown using unified.
+ */
+function toMarkdown(html) {
+  return unified()
+    .use(parse)
+    .use(rehype2remark)
+    .use(stringify)
+    .process(html);
+}
+
+/**
+ * Converts a document title into a filename.
+ */
+function namify(title) {
+  return title.toLowerCase().replace(/[ ?]/gi, "-");
```

Comment:
Consider `.trim().replace(/\s+/g, '-')` so that `foo  bar` becomes `foo-bar` instead of `foo--bar`.

Reply:
>  `foo  bar` becomes `foo-bar` instead of `foo--bar`.

As far as I can see `foo bar` becomes `foo-bar` with the existing code?

"Advanced text formatting" -> "advanced-text-formatting"

I do agree that we will probably want to think more about exactly what kind of transformation from title to filename we should have though, I am very likely missing some important cases here.

Reply:
<img width="456" alt="Screen Shot 2019-06-26 at 3 02 07 PM" src="https://user-images.githubusercontent.com/26739/60207251-6bb7dd80-9823-11e9-8f61-e96963bfebc3.png">

I only tried in the browser JS. What am I missing?

Reply:
Oh sorry, I didn't see that your `foo bar` in the original comment had multiple spaces. Sure, I can make this change.

I also wanted to replace "?". for pages like https://developer.mozilla.org/en-US/docs/Learn/HTML/Introduction_to_HTML/The_head_metadata_in_HTML.

But I think in general I didn't think much about this conversion and we will probably hit more difficult cases. Looking at it now it would probably be better to use the last part of the slug "The_head_metadata_in_HTML" rather than the title.

Reply:
I've updated this to use the slug instead of trying to namify the title, and am going to merge this now as it's got an r+ and this is the only other outstanding issue.

As I said we might want to revisit the question of filenames, but this is quite useful as it is so I'd like to get it in.